sync/podfromdu.go

```diff
@@ -38,10 +38,11 @@ var (
 )
 
 func podFromDeploymentUnit(deploymentUnit client.DeploymentSyncRequest) v1.Pod {
-	containers := []v1.Container{rancherPauseContainer}
+	var containers []v1.Container
 	for _, container := range deploymentUnit.Containers {
 		containers = append(containers, getContainer(container))
 	}
+	containers = append(containers, rancherPauseContainer)
```

Comment:
Move `rancher-pause` to the end so that exec defaults to a reasonable container. Not relevant to this PR.